### PR TITLE
Adds old hippie blood splattering

### PR DESCRIPTION
**Pull request by yoyobatty**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs )

:cl: YoYoBatty
add: Old blood hitting effects have been re-added.
/:cl:

[why]: https://imgur.com/a/aEhGEet

When you attack someone blood used to randomly fly out and hit the floor/person behind them. Somehow this got removed so I've decided to bring it back at the request of literally everyone.

### CODE REVIEW

**Comment by Manly-Dwarf** [comment]
+100
Speedmerge this nobody cares

**Comment by Maroonenahmoron** [comment]
more blood please

**Comment by Lxtoofonea** [comment]
plus one

**Comment by ghost** [comment]
minus -1

**Comment by earthmoonjune** [comment]
+1

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -1415,8 +1415,17 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 			if(prob(I.force * 2))	//blood spatter!
 				bloody = 1
 				var/turf/location = H.loc
-				if(istype(location))
-					H.add_splatter_floor(location)
```
Comment (Jujumatic) [inline code/modules/mob/living/carbon/human/species.dm:1419]: Don't straight out remove these lines if non-modular, just comment them out.

File: hippiestation/code/game/objects/effects/decals/cleanable/humans.dm [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	for(var/i in 1 to n)
+		if(!src)
+			return
+		if(splattering)
+			return
+		prev_loc = loc
+		step_towards(src,T)
+		if(!src)
+			return
+		sleep(2)
```
Comment (Jujumatic) [inline hippiestation/code/game/objects/effects/decals/cleanable/humans.dm]: ugg ugg @Carbonhell is not going to like this

File: hippiestation/code/game/objects/effects/decals/cleanable/humans.dm [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	if(splattering) return
+	if(istype(A, /obj/item))
+		var/obj/item/I = A
+		I.add_mob_blood(blood_source)
+	if(istype(A, /turf/closed/wall))
+		if(istype(prev_loc)) //var definition already checks for type
+			loc = A
+			splattering = TRUE //So "Bump()" and "Crossed()" procs aren't called at the same time
+			skip = TRUE
+			sleep(3)
```
Comment (Jujumatic) [inline hippiestation/code/game/objects/effects/decals/cleanable/humans.dm]: this one too

File: hippiestation/code/game/objects/effects/decals/cleanable/humans.dm [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+				var/obj/effect/decal/cleanable/blood/splatter/B = new(prev_loc)
+				//Adjust pixel offset to make splatters appear on the wall
+				if(istype(B))
+					B.pixel_x = dir & EAST ? 32 : (dir & WEST ? -32 : 0)
+					B.pixel_y = dir & NORTH ? 32 : (dir & SOUTH ? -32 : 0)
+			qdel(src)
+		else //This will only happen if prev_loc is not even a turf, which is highly unlikely.
+			loc = A //Either way we got this.
+			splattering = TRUE //So "Bump()" and "Crossed()" procs aren't called at the same time
+			sleep(3)
```
Comment (Jujumatic) [inline hippiestation/code/game/objects/effects/decals/cleanable/humans.dm]: fuck why are there so MANY

**Review by Jujumatic — DISMISSED** [review]
No way to modularise species.dm? Usually it's easier to comment out a TG define/proc and we make our own version, as TG changes are easier to revert if species.dm gets changed 5 years down the line

**Comment by Carbonhell** [comment]
+1

**Comment by DaydreamPuppeh** [comment]
+1

**Comment by Manly-Dwarf** [comment]
wtf merge this already you fiends

File: code/modules/mob/living/carbon/human/species.dm (hunk 1 above)
Comment (yoyobatty) [reply code/modules/mob/living/carbon/human/species.dm:1419]: It didn't get removed, it was indented further down

**Comment by yoyobatty** [comment]
I'd like to note that code came from some old hippie archive but has been modified just enough so it actually works.

**Comment by yobob591** [comment]
+1 homo

**Comment by bardofrage** [comment]
+1

**Review by GuyonBroadway — DISMISSED** [review]
Looks ok I think.

**Comment by GuyonBroadway** [comment]
Why the fuck can I not merge?

File: hippiestation/code/game/objects/effects/decals/cleanable/humans.dm [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	for(var/i in 1 to n)
+		if(!src)
+			return
+		if(splattering)
+			return
+		prev_loc = loc
+		step_towards(src,T)
+		if(!src)
+			return
+		sleep(1)
```
Comment (Carbonhell) [inline hippiestation/code/game/objects/effects/decals/cleanable/humans.dm]: >sleep
This doesn't look ok

**Review by Carbonhell — DISMISSED** [review]

**Review by GuyonBroadway — DISMISSED** [review]
Handle the sleeps to avoid the big oof

**Comment by yoyobatty** [comment]
Working on it boys

**Comment by GuyonBroadway** [comment]
At the risk of suggesting feature creep why not get it to apply a few dozen units of blood reagent to the floor so extended fistfights can result in a literal bloodbath.

**Comment by yoyobatty** [comment]
This only affects when you attack with items, not melee, sorry.

**Comment by DaydreamPuppeh** [comment]
Then we should make brass knuckles

**Pull request merged**